src/main/UserActivityMonitor.js

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2015-2016 Yuya Ochiai
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+'use strict';
+
+import EventEmitter from 'events';
+
+import electron from 'electron';
+
+const {app} = electron;
+
+/**
+ * Monitors system idle time, listens for system events and fires status updates as needed
+ */
+export default class UserActivityMonitor extends EventEmitter {
+  constructor() {
+    super();
+
+    this.isActive = true;
+    this.forceUserIsInactive = false;
+    this.idleTime = 0;
+    this.lastStatusUpdate = 0;
+    this.systemIdleTimeIntervalID = -1;
+
+    this.config = {
+      regularUpdateFrequencyMs: 1 * 1000,
+      statusUpdateThresholdMs: 5 * 1000,
+      userActivityTimeoutSec: 5 * 60,
+    };
+
+    this.handleSuspend = this.handleSuspend.bind(this);
+    this.handleResume = this.handleResume.bind(this);
+    this.handleShutdown = this.handleShutdown.bind(this);
+    this.handleLockScreen = this.handleLockScreen.bind(this);
+    this.handleUnlockScreen = this.handleUnlockScreen.bind(this);
+  }
+
+  get userIsActive() {
+    return this.forceUserIsInactive ? false : this.isActive;
```

Comment:
0/5 since it's totally nitpicking code styles, but I'd tend towards doing this without a ternary statement like
```suggestion
    return this.isActive && !this.forceUserIsInactive;
```